### ISSUE TITLE
Getting "B1" is not in list error with 0.4.0 release for Landsat8 dataset 

### ISSUE THREAD

**Issue by TolgaAktas** [microsoft/torchgeo#1134]
### Issue

Previously, the following instantiation used to work with 0.3.1 version:
```python
dataset = Landsat8(root=root_path,bands=["B1","B2","B3","B4","B5","B6","B7","B8","B9","B10", "B11","QA_PIXEL"])
```
But with 0.4.0, I am getting the following error stack: 
```
Traceback (most recent call last):
  File "/home/ta6167/diffusion_experiments/improved_ddpm_experiments/main.py", line 14, in <module>
    dataset = Landsat8(root=root_path,bands=["B1","B2","B3","B4","B5","B6","B7","B8","B9","B10", "B11","QA_PIXEL"])
  File "/home/ta6167/.local/lib/python3.10/site-packages/torchgeo/datasets/landsat.py", line 80, in __init__
    super().__init__(root, crs, res, bands, transforms, cache)
  File "/home/ta6167/.local/lib/python3.10/site-packages/torchgeo/datasets/geo.py", line 371, in __init__
    band_indexes = [self.all_bands.index(i) + 1 for i in bands]
  File "/home/ta6167/.local/lib/python3.10/site-packages/torchgeo/datasets/geo.py", line 371, in <listcomp>
    band_indexes = [self.all_bands.index(i) + 1 for i in bands]
ValueError: 'B1' is not in list
```
I am guessing the designation for channel names might have changed, what is the correct way to provide the channel names for 0.4.0 version?

**Comment by adamjstewart** [comment]
Thanks for reporting this bug! This appears to be an unintended consequence of @ashnair1's #687. Let me see if I can come up with a bug fix and we'll fix this in the 0.4.1 release (due in 1–2 weeks).

**Comment by adamjstewart** [comment]
Should be fixed in #1135!
